src/provider/application.js

```diff
@@ -240,14 +249,17 @@ class Application extends EventEmitter {
 
     if (message) {
       logger.log('>> provider', this.acls, message);
+
+      const dispatch = this.dispatchFunction || parent.postMessage;
```

Comment:
move this logic into the init block. It should also contain the logic to skip messages when not embedded rather than coupling that logic to custom dispatch methods.
```js
this.dispatchFunction = dispatchFunction || (message, targetOrigin) => {
    if (window.self !== window.top) {
     parent.postMessage(message, targetOrigin);
   }
 });
``

Reply:
Moved: 7968afd